pkg/server/service/service.go

```diff
@@ -161,9 +161,24 @@ func (m *Manager) getWRRServiceHandler(ctx context.Context, serviceName string,
 		if err != nil {
 			return nil, err
 		}
-
 		balancer.AddService(service.Name, serviceHandler, service.Weight)
+		if config.HealthCheck == nil {
+			continue
+		}
+		childName := service.Name
+		updater, ok := serviceHandler.(healthcheck.StatusUpdater)
+		if !ok {
+			log.FromContext(ctx).Debugf("Child service %v will not propagate to its parent %v any status change", service.Name, serviceName)
+			continue
+		}
+		log.FromContext(ctx).Debugf("Child service %v will update parent %v on status change", service.Name, serviceName)
+		// TODO(mpl): do we make RegisterStatusUpdater return an error (e.g. in case the
+		// updater actually does not want healthcheck enabled, according to its own config)
+		updater.RegisterStatusUpdater(func(up bool) {
+			balancer.SetStatus(serviceName, childName, up)
```

Comment:
```suggestion
			balancer.SetStatus(serviceName, service.Name, up)
```

Reply:
nope, it's very important to not change that part. I will let you think about it for a while ;)